crates/bevy_reflect/src/serde/de.rs

```diff
@@ -1,9 +1,17 @@
 use crate::{
```

Comment:
> > This will almost certainly require a brand new set of serializers and deserializers. Doing this would have been true for any impl (ex: we don't want to store the full type_name for each instance of a type), so I think its safe to say that this scenario is orthogonal.

Yeah, based on other discussions it seems their intentions are at odds: these de/serializers are meant to be human-readable and easily modified, while the binary formats are designed for efficiency and compactness.

> It would mean that by default we only serialize component fields users have manually configured in their scene (ex: setting the entity position to something other than zero).

I understand the idea behind this, but I'm curious what that implementation would look like. How do we determine what fields are "set" vs "unset"? Would we need to run a comparison on each field to its default value? And how would we know it impls `Default`?

> From my perspective, this implementation is compatible with our future plans.

Awesome! The intention was to keep things relatively non-controversial/future-oriented while still cleaning up the format a fair amount.

I suppose this means that we're choosing this format over #4561 then?

Reply:
> I understand the idea behind this, but I'm curious what that implementation would look like. How do we determine what fields are "set" vs "unset"? Would we need to run a comparison on each field to its default value? And how would we know it impls Default?

I think we'll want something along the lines of "reflect diffing", which I implemented (but never merged) super early on in Bevy here: #944, for exactly this purpose.

Basically call `component_instance.diff(&default_component_instance)`. What "default" means is up debate. Maybe it is literally the `Default` trait, maybe it is `FromWorld`, maybe we also support manually defined DynamicStructs for special cases.

How this "default component instance" is stored and accessed is also up for debate. Maybe we do it on the spot each time, maybe we maintain a "database" of them in TypeRegistry.  

> I suppose this means that we're choosing this format over https://github.com/bevyengine/bevy/pull/4561 then?

Yup it has my preference!

Reply:
> > I suppose this means that we're choosing this format over #4561 then?
> 
> Yup it has my preference!

Sounds good 🙂

I'll close it then and direct others to this one instead.